File: Examples/CSharp/AsposePDF/Links-Actions/ExtractLinks.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.InteractiveFeatures.Annotations;
using System.Collections;

namespace CSharp.AsposePdf.LinksActions
{
    public class ExtractLinks
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_LinksActions();


            //Open document
            Document document = new Document(dataDir+ "ExtractLinks.pdf");

            //Extract actions
            Page page = document.Pages[1];
            AnnotationSelector selector = new AnnotationSelector(new LinkAnnotation(page, Aspose.Pdf.Rectangle.Trivial));
            page.Accept(selector);
            IList list = selector.Selected;

            Annotation annotation = (Annotation)list[0];

            //Save updated document
            document.Save(dataDir + "ExtractLinks_out.pdf");
            
        }
    }
}

File: Examples/VisualBasic/AsposePDF/Links-Actions/ExtractLinks.vb
Imports System.IO
Imports System
Imports Microsoft.VisualBasic
Imports Aspose.Pdf
Imports Aspose.Pdf.Annotations
Imports System.Collections

Namespace AsposePDF.LinksActions
    Public Class ExtractLinks
        Public Shared Sub Run()
            ' ExStart:ExtractLinks
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_LinksActions()
            ' Open document
            Dim document As New Document(dataDir & Convert.ToString("ExtractLinks.pdf"))
            ' Extract actions
            Dim page As Page = document.Pages(1)
            Dim selector As New AnnotationSelector(New LinkAnnotation(page, Rectangle.Trivial))
            page.Accept(selector)
            Dim list As IList = selector.Selected
            Dim annotation As Annotation = DirectCast(list(0), Annotation)
            dataDir = dataDir & Convert.ToString("ExtractLinks_out_.pdf")
            ' Save updated document
            document.Save(dataDir)
            ' ExEnd:ExtractLinks
            Console.WriteLine(Convert.ToString(vbLf & "Links extracted successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace
